### PR TITLE
Add Homepage Design and Quiz Category Selection 

### PR DESCRIPTION
**Pull request by Kendrick-Oppong**
This PR introduces the following changes:

**Created HomeComponent:**
1. Displays a list of quiz categories (HTML, CSS, JavaScript, Accessibility) with corresponding icons.
2. Uses an EventEmitter to emit the selected category title to the parent component.

**Updated AppComponent:**
1. Injected the QuizService to retrieve quiz data.
2. Defined a property to store the selected quiz category.
3. Initialized the category title property in AppComponent for future dynamic component rendering.

**HTML/CSS Modifications:**

1. Updated the app.component.html and home.component.html templates to reflect the new structure.
2. Added UI elements and visuals for each quiz category.

Here is the UI 
QuizApp-HOMEPAGE

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
-  imports: [RouterOutlet],
+  imports: [HomeComponent],
   templateUrl: './app.component.html',
-  styleUrl: './app.component.css'
+  styleUrl: './app.component.css',
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
   title = 'quiz-app';
+  selectedCategory: string = '';
```
Comment (amt-augustineopokukesse) [inline src/app/app.component.ts]: Adding the string type in this instance is verbose.. If you initialise a property, as in this case an empty string, TypeScript infers the type from the initialization

File: src/app/components/home/home.component.css [hunk 2]
```diff
@@ -0,0 +1,39 @@
+h1 {
+  font-weight: 100;
+}
+
+p {
+  margin-top: 2rem; 
+  color: var(--grey-navy);
+  font-style: italic;
+  font-size: 1.25rem; /* 20px*/
```
Comment (amt-augustineopokukesse) [inline src/app/components/home/home.component.css]: Why the comment on this line?

File: src/app/components/home/home.component.css [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+ul div {
+  display: flex;
+  align-items: center;
+  gap: 2rem;
+  width: 100%;
+  background-color: var(--pure-white);
+  padding: 1.25rem; /* 20px*/
+  border-radius: 1.5rem; /* 24px*/
```
Comment (amt-augustineopokukesse) [inline src/app/components/home/home.component.css]: remove the comments please

Comment (Kendrick-Oppong) [reply src/app/components/home/home.component.css]: OK

**Review by amt-augustineopokukesse — CHANGES_REQUESTED** [review]
Address the minor changes

File: src/app/app.component.ts (hunk 1 above)
Comment (Kendrick-Oppong) [reply src/app/app.component.ts]: OK

**Review by amt-augustineopokukesse — APPROVED** [review]

**Pull request merged**